lib/token.h

```diff
@@ -88,6 +89,9 @@ struct TokenImpl {
     /** Bitfield bit count. */
     unsigned char mBits;
 
+	// Pointer to a template in the template simplifier
+    TemplateSimplifier::TokenAndName* mTemplateSimplifierPointer;
```

Comment:
A token can be in multiple containers.  TokenAndName is a terrible data structure. We really need different ones for declarations and instantiations.  We spend a lot of time recalculating things that could be cached in these structures.

A vector of std::functions would be better that would be called when a token is destroyed.

Reply:
In that case I will modify this change.